lib/shopify_api/plugs/customer_authenticator.ex

```diff
@@ -150,13 +150,4 @@ defmodule ShopifyAPI.Plugs.CustomerAuthenticator do
 
   defp handle_error(conn, {:error, _}),
     do: send_unauthorized_response(conn, "Could not parse auth_payload")
-
-  defp handle_error(conn, {:error, _, _}),
-    do: send_unauthorized_response(conn, "Could not parse auth_payload")
-
-  defp handle_error(conn, _error) do
-    conn
-    |> resp(500, "Unhandled Error")
-    |> halt()
-  end
```

Comment:
Dialyzer complained about these not being possible matches once I made the refactor, as with the other `case` branch in `ShopifyAPI.GraphQL.Response`. I've removed them accordingly.